### PR TITLE
Yaw issue25 complete profile page

### LINKED ISSUE #25: Complete Profile Page
Page Url: _none_ , user will be redirected to this page if they `isAuthenticated && !isProfileComplete` 
Module: Auth

Components: (if any)
- form

Usage: User is required to complete their profile before using the app.

# Banner
Yellow Banner: Please complete your profile before proceeding

# Validation
## Name Field _(Text)_
- required

## Gender Field _(Dropdown)_
- required

## Phone Number Field _(Number)_
- required
- min length 7
- only number

## Date of Birth Field _(Date Picker)_
- required

## Faculty Field _(Dropdown)_
- required

## Course Program Field _(Dropdown)_
- required

## Intake Session Field _(Dropdown)_
- required

## Intake Semester Field _(Dropdown)_
- required

## Submit Button
- button will be disabled onValidSubmit, to prevent user submit more than once

# onSubmit
## Fail
- show Error Toast

## Success
- show Success Toast
- redirect to the home page

# Progress
- [x] Development
- [ ] Unit Test

### PR DESCRIPTION
**Pull request by HowardYaw**
resolved #25 

image

### CODE REVIEW

**Review by kachun333 — DISMISSED** [review]

File: src/main/webapp/app/modules/auth/complete-profile/complete-user-profile.tsx [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+                <AvGroup>
+                  <AvField
+                    id="phoneNumber"
+                    label={translate('clubmanagementApp.userProfile.phoneNumber')}
+                    type="text"
+                    name="phoneNumber"
+                    placeholder="E.g 123857221"
+                    validate={{
+                      required: { value: true, errorMessage: 'Please enter your Phone Number' },
+                      pattern: { value: '^([1-9][0-9]{6,20})$', errorMessage: 'Please enter a valid Phone Number without Country Code' }
```
Comment (CarlosSia) [inline src/main/webapp/app/modules/auth/complete-profile/complete-user-profile.tsx]: https://en.wikipedia.org/wiki/Telephone_numbers_in_Malaysia
malaysia phone number at most 10 digit but the regex allow up to 20 digits

File: src/main/webapp/i18n/zh-cn/userUniInfo.json [hunk 2]
```diff
@@ -2,28 +2,35 @@
   "clubmanagementApp": {
     "userUniInfo": {
       "home": {
-        "title": "User Uni Infos",
-        "createLabel": "创建新 User Uni Info",
+        "title": "个人大学资料",
+        "createLabel": "创建新个人大学资料",
+        "editLabel": "更新新个人大学资料",
```
Comment (CarlosSia) [inline src/main/webapp/i18n/zh-cn/userUniInfo.json]: is it better to use 更新个人大学资料? because if it is updating it is not new anymore

Comment (HowardYaw) [reply src/main/webapp/i18n/zh-cn/userUniInfo.json]: resolved in b24b1b151c6bb1e741153514eec75a336c4239fa

File: src/main/webapp/app/modules/auth/complete-profile/complete-user-profile.tsx (hunk 1 above)
Comment (HowardYaw) [reply src/main/webapp/app/modules/auth/complete-profile/complete-user-profile.tsx]: resolved in b24b1b151c6bb1e741153514eec75a336c4239fa

**Review by kachun333 — APPROVED** [review]

**Review by LUXIANZE — APPROVED** [review]

**Pull request merged**